test/tests/preconditioners/reuse/tests

```diff
@@ -0,0 +1,20 @@
+[Tests]
+  issues = '#21868'
+  design = 'NonlinearSystem.md'
+  [without_reuse]
+    type = CSVDiff
+    input = convergence.i
+    csvdiff = base_case.csv
+    requirement = 'Convergence matches previous version of MOOSE without the '
+                  'preconditioner reuse system'
+  []
+  [with_reuse]
+    type = CSVDiff
+    input = convergence.i
+    cli_args = "Executioner/reuse_preconditioner=true Outputs/csv/file_base=reuse_case"
+    csvdiff = reuse_case.csv
+    requirement = 'Preconditioner is reused until the linear iterations exceed '
+                  'the value of reuse_preconditioner_max_its upon which the '
+                  'system recalculates the preconditioner'
```

Comment:
> Since the example uses NEWTON, the number of linear iterations is just a proxy for number of Newton iterations?

No, the number of linear iterations should be, well... the number of iterations whichever linear solver uses to solve the linear system of equations. In this case the linear solver is gmres IIRC.

Reply:
With this option for Newton we are still solving the nonlinear system using the full, "exact" Newton-Raphson method and we're still solving the exact linearized system.  However, our method of solving the linearized system is going to get less and less effective as we reuse a prior preconditioner, hence the heuristic based on the number of linear iterations.  This all works because forming the preconditioner is the most expensive part of this whole process.

For PJFNK preconditioner reuse will work (you can use the same set of options) but I'm not sure how effective it will be.  You're still forming a preconditioner so this has the potential to reduce the amount of work needed to solve the problem but whether or not it's effective is going to depend on the cost of forming the preconditioner versus the cost of iteration.

Reply:
> In this case the linear solver is gmres IIRC.

I'll give you that, dumb question. I was thinking direct solve... 

@reverendbedford Thanks for the details: Does the number of linear iterations accumulate across Newton iterations or it will always compute the preconditioner upon starting a new one? (You may have already explained this but I can't see it in the docs)

Reply:
Okay, it seems carryover only happens within the Newton solve according to the parameter description.